### PR TITLE
feat: Tavily search integration

### PR DESCRIPTION
**Pull request by dxmaptin**
## Description

Implemented the Tavily search functionality and completed local testing with successful results.

## Motivation and Context

This change introduces the search feature to the Tavily API. It addresses the need for efficient search capabilities. This feature enhances the user experience by enabling faster, more relevant search results.

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ x] Integrated search logic
- [ x] Added unit tests for search functionality
- [ x] Performed successful local testing

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/toolkits/search_toolkit.py [hunk 1]
```diff
@@ -305,7 +305,86 @@ def query_wolfram_alpha(self, query: str, is_detailed: bool) -> str:
...
+            return [{"error": "Missing API key."}]
+        except InvalidAPIKeyError:
+            return [{"error": "Invalid API key."}]
+        except UsageLimitExceededError:
+            return [
+                {
+                    "error": "Usage limit exceeded. Check your plan's "
+                    "usage limits."
+                }
+            ]
```
Comment (Wendong-Fan) [inline camel/toolkits/search_toolkit.py:385]: this is great to handle specific error, but from code concise and maintain's perspective, I would suggest just use 
```
        except Exception as e:
            return [{"error": f"An unexpected error occurred: {e!s}"}]
```
to catch all exceptions

File: camel/toolkits/search_toolkit.py [hunk 2]
```diff
@@ -305,7 +305,86 @@ def query_wolfram_alpha(self, query: str, is_detailed: bool) -> str:
...
+            {
+                'result_id': 1,
+                'title': 'OpenAI',
+                'description': 'An AI research organization...',
+                'long_description': 'OpenAI is a non-profit...',
+                'url': 'https://www.openai.com',
+                'content': 'OpenAI focuses on AI safety...',
+                'images': ['https://image.url/1.jpg'],
+                'published_date': '2024-09-15'
+            }
```
Comment (Wendong-Fan) [inline camel/toolkits/search_toolkit.py:354]: the output example should be under `Returns` part

File: camel/toolkits/search_toolkit.py [hunk 3]
```diff
@@ -305,7 +305,86 @@ def query_wolfram_alpha(self, query: str, is_detailed: bool) -> str:
...
+                - topic (str): The search category, e.g., "general" or "news."
+                - days (int): Time frame in days for news-related searches.
+                - max_results (int): Max number of results to return
+                  (overrides `num_results`).
+                See https://docs.tavily.com/docs/python-sdk/tavily-search/
+                api-reference for details.
+
+        Returns:
+            List[Dict[str, Any]]: A list of dictionaries representing search
+            results. Each dictionary contains:
```
Comment (Wendong-Fan) [inline camel/toolkits/search_toolkit.py:332]: docstring format

**Review by Wendong-Fan — COMMENTED** [review]
Thanks @dxmaptin ! Left some comments below, could we also support tavily extract further?
https://docs.tavily.com/docs/python-sdk/tavily-extract/getting-started

**Comment by dxmaptin** [comment]
Fixed all issues raised above

**Pull request merged**